modules/launcher/src/main/java/org/mule/runtime/module/launcher/DefaultMuleContainer.java

```diff
@@ -270,27 +281,25 @@ public void start(boolean registerShutdownHook) throws MuleException {
       coreExtensionManager.setServiceRepository(serviceManager);
       coreExtensionManager.setTroubleshootingService(troubleshootingService);
 
-      validateLicense();
-      showSplashScreen();
+      // Waits for all bootstrapping configurations to be ready before progressing any further
+      if (!configurationsReadyBarrier.getAsBoolean()) {
```

Comment:
why not use a latch instead of a supplier? i think the code would be clearer with a latch